src/stream/src/executor/hash_join.rs

```diff
@@ -1026,6 +1197,13 @@ impl<K: HashKey, S: StateStore, const T: JoinTypePrimitive> HashJoinExecutor<K,
                         }
                         // Insert back the state taken from ht.
                         side_match.ht.update_state(key, matched_rows);
+                        for matched_row in matched_rows_to_clean {
+                            if side_match.need_degree_table {
+                                side_match.ht.delete(key, matched_row);
+                            } else {
+                                side_match.ht.delete_row(key, matched_row.row);
+                            }
+                        }
```

Comment:
To ensure that the state cleaning works properly, could you please add a unit test for the interval join?

Reply:
> To ensure that the state cleaning works properly, could you please add a unit test for the interval join?

fixed